In [5]:
import pandas as pd
from datatable import dt, f

TODO: Deixar aqui apenas os dados brutos

# Aquisição dos dados brutos

Primeiramente, conforme falado na seção anterior, unimos os microdados de todos os anos para ter um olhar menos fechado em um único ano.

In [9]:
def get_path_table_by_year(year):
  path_table = '/'.join([
    '../data/raw',
    'censo_{year}.zip',
    'Microdados do Censo da Educaç╞o Superior {year}',
    'dados',
    'MICRODADOS_CADASTRO_CURSOS_{year}.CSV'
  ]).format(year = year)
  return path_table

In [10]:
def aggregate_raw_data(from_year, to_year):
  for y in range(from_year, to_year + 1):
    path_table = get_path_table_by_year(y)
    dataset = dt.fread(path_table)
  return dataset

In [11]:
df_raw = aggregate_raw_data(2010, 2019)
display(df_raw)

<Frame#7fbdd3eb9e70 246081x196>

Para salvar esses dados agregados em um csv, é necessário fazer o seguinte:

In [19]:
df_raw.to_csv('../data/raw/aggregated_dataset.csv')

Após a agregação, a tabela bruta para trabalho ficou com 246,081 linhas e 196 colunas.  
Após isso, era necessário entender mais sobre as principais características e atributos da base que nos seriam interessantes.